service/resource/vaultcrt/create_test.go

```diff
@@ -128,3 +129,50 @@ func Test_Resource_VaultCrt_newCreateChange(t *testing.T) {
 		}
 	}
 }
+
+func Test_Resource_VaultCrt_ensureVaultRole_Organizations_intact(t *testing.T) {
+	testCases := []struct {
+		CustomObject          certificatetpr.CustomObject
+		ExpectedOrganizations []string
+	}{
+		// Test 0 ensures a non-nil current state results in the create state to be
+		// empty.
+		{
+			CustomObject: certificatetpr.CustomObject{
+				Spec: certificatetpr.Spec{
+					ClusterID:        "foobar",
+					ClusterComponent: "api",
+					Organizations:    []string{"system:masters"},
+				},
+			},
+			ExpectedOrganizations: []string{"api", "system:masters"},
+		},
+	}
+
+	var err error
+	var newResource *Resource
+	{
+		resourceConfig := DefaultConfig()
+
+		resourceConfig.K8sClient = fake.NewSimpleClientset()
+		resourceConfig.Logger = microloggertest.New()
+		resourceConfig.VaultCrt = vaultcrttest.New()
+		resourceConfig.VaultRole = vaultroletest.New()
+
+		newResource, err = New(resourceConfig)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+
+	for i, tc := range testCases {
+		err := newResource.ensureVaultRole(tc.CustomObject)
```

Comment:
For my interest, does the test fail when you remove your actual fix from `key.Organizations`? Why do we need to test `VaultCrt.ensureVaultRole` when the fix is in `key.Organizations`? I thought the test could be something like this. 

```
check
key.Organizations(tc.CustomObject)
check
key.Organizations(tc.CustomObject)
check
key.Organizations(tc.CustomObject)
check
```